src/NuGet.Clients/PackageManagement.VisualStudio/EnvDTEProjectAdapter.cs

```diff
@@ -159,18 +162,23 @@ public bool IsLegacyCSProjPackageReferenceProject
                     // A legacy CSProj can't be CPS, must cast to VSProject4 and *must* have at least one package
                     // reference already in the CSProj. In the future this logic may change. For now a user must
                     // hand code their first package reference. Laid out in longhand for readability.
-                    if (AsIVsHierarchy?.IsCapabilityMatch("CPS") ?? true)
-                    {
-                        _isLegacyCSProjPackageReferenceProject = false;
-                    }
-                    else if (AsVSProject4 == null ||
-                        (AsVSProject4.PackageReferences?.InstalledPackages?.Length ?? 0) == 0)
+                    if (AsIVsHierarchy?.IsCapabilityMatch("CPS") ?? true || AsVSProject4 == null)
                     {
                         _isLegacyCSProjPackageReferenceProject = false;
                     }
                     else
                     {
-                        _isLegacyCSProjPackageReferenceProject = true;
+                        var restoreProjectStyle = GetMSBuildProperty(AsIVsBuildPropertyStorage, RestoreProjectStyle, null);
+
+                        if (!restoreProjectStyle?.Equals(ProjectStyle.PackageReference.ToString(), StringComparison.OrdinalIgnoreCase) ?? true
+                            && (AsVSProject4.PackageReferences?.InstalledPackages?.Length ?? 0) == 0)
```

Comment:
1 is covered above: if (AsIVsHierarchy?.IsCapabilityMatch("CPS") ?? true || AsVSProject4 == null)
Do you have specific tests you would like me to run to verify that everything works correctly?